lib/beaker/host/mac/pkg.rb

```diff
@@ -118,7 +118,15 @@ def pe_puppet_agent_promoted_package_info( puppet_collection = nil, opts = {} )
 
     variant, version, arch, codename = self['platform'].to_array
     release_file = "/repos/apple/#{version}/#{puppet_collection}/#{arch}/puppet-agent-*"
-    download_file = "puppet-agent-#{variant}-#{version}.tar.gz"
+
+    # macOS puppet-agent tarballs haven't always included arch
+    agent_version = opts[:puppet_agent_version]
+    download_file = if agent_version && (agent_version.to_f < 6.28 || agent_version.to_f < 7.18)
```

Comment:
From what I can tell, this method is only ever called from beaker-pe and `puppet_agent_version` is always passed as an option. It starts in beaker-pe, where we call [`get_puppet_agent_version`](https://github.com/puppetlabs/beaker-pe/blob/fda23282928edcbc11af0c572e8f7b697e9634a6/lib/beaker-pe/install/pe_utils.rb#L892)

That method either [raises or returns a non-nill value](https://github.com/puppetlabs/beaker-pe/blob/fda23282928edcbc11af0c572e8f7b697e9634a6/lib/beaker-pe/install/pe_utils.rb#L1106-L1109)

Then we call [install_puppet_agent_pe_promoted_repo_on](https://github.com/puppetlabs/beaker-pe/blob/fda23282928edcbc11af0c572e8f7b697e9634a6/lib/beaker-pe/install/pe_utils.rb#L899)

That method is [implemented in beaker-puppet](https://github.com/voxpupuli/beaker-puppet/blob/4d6d47e842b68250ea9f86c8bdeda979ff2e52fe/lib/beaker-puppet/install_utils/foss_utils.rb#L1367) Not sure why it's in beaker-puppet (!?)

And that calls [Beaker::Host#pe_puppet_agent_promoted_package_info](https://github.com/voxpupuli/beaker-puppet/blob/4d6d47e842b68250ea9f86c8bdeda979ff2e52fe/lib/beaker-puppet/install_utils/foss_utils.rb#L1384) which is the method this PR changes.